openvdb/tools/FastSweeping.h

```diff
@@ -0,0 +1,1432 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+//
+/// @file    FastSweeping.h
+///
+/// @author  Ken Museth
+///
+/// @brief   Defined the six functions {fog,sdf}To{Sdf,Ext,SdfAndExt} in
+///          addition to the two functions maskSdf and dilateSdf. Sdf denotes
+///          a signed-distance field (i.e. negative values are insdie), fog
+///          is a scalar fog volume (i.e. higher values are inside), and Ext is
+///          a field (currently limited to a scalar) that is extended off the
+///          iso-surface. All these functions are implemented by the methods of
+///          the class dubbed FastSweeping.
+///
+/// @todo    1) Sort uint32 offsets instead of array of Coord
+///          2) Concurrent bi-directional sweeping
+///          3) Rebuild narrow-band level set
+///          4) Allow for the grid types of the sdf and ext to be different
+///
+/// @note    Solves the (simplified) Eikonal Eq: @f$|\nabla \phi|^2 = 1@f$ and
+///          performs velocity extension,  @f$\nable f\nabla \phi = 0@f$, both
+///          by means of the fast sweeping algorithm detailed in:
+///          "A Fast Sweeping Method For Eikonal Equations"
+///          by H. Zhao, Mathematics of Computation, Vol 74(230), pp 603-627, 2004
+///
+/// @details The algorithm used below for parallel fast sweeping was first publised in:
+///          "New Algorithm for Sparse and Parallel Fast Sweeping: Efficient
+///          Computation of Sparse Distance Fields" by K. Museth, ACM SIGGRAPH Talk,
+///          2017, http://www.museth.org/Ken/Publications_files/Museth_SIG17.pdf
+
+#ifndef OPENVDB_TOOLS_FASTSWEEPING_HAS_BEEN_INCLUDED
+#define OPENVDB_TOOLS_FASTSWEEPING_HAS_BEEN_INCLUDED
+
+//#define BENCHMARK_FAST_SWEEPING
```

Comment:
Perhaps a better solution here would be to allow the enabling of the `CpuTimer` wholescale. i.e. build openvdb with `-DENABLE_PROFILING` or some such and, in `CpuTimer`, you do something like:

```
#ifdef ENABLE_PROFILING
// normal implementation of timer
#else 
struct CpuTimer { CpuTimer(std::string&) { /*noop*/} ... };
#endif

This would help clear up the code somewhat

Reply:
I'm planning to remove all the timing for the final commit - it is, however, useful as long as I'm making changes to the code.

Reply:
I absolutely agree - which is why it would actually be nice if, in theory, these could be left in. I'm basically proposing that we introduce a system where it's encourages to add timings around code and enable/disable these metrics wholesale at compile time across VDB. It's a passing thought which, so long as these are being removed, doesn't apply to this PR.